firedrake/utils.py

```diff
@@ -6,9 +6,11 @@
 from pyop2.datatypes import RealType     # noqa: F401
 from pyop2.datatypes import IntType      # noqa: F401
 from pyop2.datatypes import as_ctypes    # noqa: F401
+from pyop2.mpi import COMM_WORLD
 from firedrake_configuration import get_config
 
-_current_uid = 0
+# MPI key value for storing a per communicator universal identifier
+FIREDRAKE_UID = COMM_WORLD.Create_keyval()
```

Comment:
```suggestion
from mpi4py import MPI
...
FIREDRAKE_UID = MPI.Comm.Create_keyval()
```

Then it's clear that the keyval is not specific to a comm?